### ISSUE TITLE
Block chooser doesn't show enough options on initial view

### ISSUE THREAD

**Issue by nicklee** [wagtail/wagtail#9754]
### Is your proposal related to a problem?

The block chooser consistently has a vertical scrollbar which makes choosing a block more difficult than it needs to be, as the user isn't aware of what is below the fold (unless they are **very** familiar with the system).

The search option that is shown at the top can significantly improve the speed of adding a block, however without having pre-existing knowledge of what the avaliable blocks are named, it's not great for new users of the system.

Existing example:

![Chooser fly out](https://user-images.githubusercontent.com/298766/205892816-d911ca44-4f89-45dd-b197-388837811444.png)

### Describe the solution you'd like

With the above in mind, I'd suggest optimising the chooser to ensure that more blocks are avaliable without the need to scroll. Initial thoughts are:

1. Introducing multiple columns
2. Reducing the amount of space between items
3. Refining the category titles
4. Giving the items a more sensible order (Alphabetical?)
5. Reducing the h2-h5 options into a single heading, and relying on the formatting options within the RTE to apply h2-h5

**Comment by nicklee** [comment]
I've attached some mockups illustrating how the suggested solutions could potentiallly work:

Single column that just makes better use of the space, meaning more items are displayed initially without the need to scroll:
![single-col](https://user-images.githubusercontent.com/298766/205928568-551bbca9-d75e-47f4-ae8c-5d7553e99854.png)

Double column, which maximises even more so, without covering too much more of the RTE below:
![two-col](https://user-images.githubusercontent.com/298766/205928577-9c272a2e-1e0f-40b3-a932-0df61dbfb0ad.png)

On both versions I've removed the heading from the initial group of inline/formatting blocks, rationale being that users are typically familiar with these items (and if custom ones are added their training should cover these).

I've also moved to ordering the blocks within each group alphabetically, as this can aid findability when you're scanning a list. I appreciate that some editors may want to display their most used blocks first, however, I'm hoping that now more, if not all, blocks are displayed then alphabeticalising will be the logical way forward.
